### PR TITLE
[DB Import] Add first SQL tests and bigquery emulator plumbing

### PR DESCRIPTION
**Pull request by beckerhe**
This is adding a context manager which can start the bigquery emulator for local testing.

It also adds tests for the helper functions that we have in db.py

### CODE REVIEW

File: devtools/db_import/db_import/db_test.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+from google.cloud import bigquery
+
+from db_import import db
+from db_import import bigquery_emulator
+
+
+@unittest.skipUnless(bigquery_emulator.is_bigquery_emulator_available(
+), "The BigQuery emulator is not available on this machine. Check out https://github.com/goccy/bigquery-emulator for more details."
+                    )
```
Comment (pzread) [inline devtools/db_import/db_import/db_test.py:18]: The auto-format behaves strangely here. I'm wondering if the trick below will make it better (adding a comma at the end)

```suggestion
@unittest.skipUnless(
  bigquery_emulator.is_bigquery_emulator_available(),
  "The BigQuery emulator is not available on this machine. Check out https://github.com/goccy/bigquery-emulator for more details.",
)
```

File: devtools/db_import/db_import/bigquery_emulator.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  http_port = _reserve_local_port()
+  grpc_port = _reserve_local_port()
+
+  with subprocess.Popen([
+      "bigquery-emulator", f"--project={project_name}",
+      f"--dataset={dataset_name}", f"--port={http_port}",
+      f"--grpc-port={grpc_port}"
+  ],
+                        stdout=subprocess.DEVNULL,
+                        stderr=subprocess.DEVNULL) as process:
```
Comment (pzread) [inline devtools/db_import/db_import/bigquery_emulator.py:70]: Same here, adding a comma at the end might make auto-format look better.
```suggestion
                        stderr=subprocess.DEVNULL,) as process:
```

**Review by pzread — APPROVED** [review]
This is very cool!

File: devtools/db_import/db_import/db_test.py (hunk 1 above)
Comment (beckerhe) [reply devtools/db_import/db_import/db_test.py]: Yes. that works. Fixed. Thx.

**Pull request merged**